### PR TITLE
refactor: sign back flow

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- UI changes relating to the sign back flow.
- The buttons are now all white.
- Display user avatar of the previously logged user.

Preview:

https://github.com/dailydotdev/apps/assets/13744167/33cb9b1f-a3b1-42f6-b9d3-d4ae9f4ad127

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1791 #done

### CODE REVIEW

File: packages/shared/src/styles/components/buttons.css
Comment (sshanzel) [inline packages/shared/src/styles/components/buttons.css:265]: Tried to do it using the Button component, but there were **a lot** of overriding classes and wrapping of components. Which is way more complicated than creating one for the sign back in button.

File: packages/shared/src/components/auth/AuthContainer.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthContainer.tsx:5]: This one is used quite often right, just something to be aware of :)

File: packages/shared/src/components/auth/AuthDefault.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthDefault.tsx:153]: This will be a bit refactored with my changes I think :)

File: packages/shared/src/components/auth/AuthModalFooter.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthModalFooter.tsx:26]: I think this one I also modified completly

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthOptions.tsx:198]: Why do we need both this one as the onValidRegistration?
Seems like we could use one right?

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthOptions.tsx:287]: Oh and here as well.

File: packages/shared/src/components/buttons/Button.tsx
Comment (rebelchris) [inline packages/shared/src/components/buttons/Button.tsx:136]: This one feels a bit weird, is it really needed?

File: packages/shared/src/styles/components/buttons.css
Comment (rebelchris) [reply packages/shared/src/styles/components/buttons.css:265]: Hmm that's sad :( 
Maybe once we refactor the button

**Review by rebelchris — COMMENTED** [review]
Overall there might be some changes with my branch, I'll try and merge ASAP.
But let's see might make things easier for you actually.

File: packages/shared/src/styles/components/buttons.css
Comment (sshanzel) [reply packages/shared/src/styles/components/buttons.css:265]: Yeah, hopefully soon, then we can also do the input elements. Lots of things are going on in those components.

File: packages/shared/src/components/buttons/Button.tsx
Comment (sshanzel) [reply packages/shared/src/components/buttons/Button.tsx:136]: Oh, this was some refactored down code. Forgot to remove the other condition.

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthOptions.tsx:287]: Yep, we need these calls as our flow sometimes goes through registration flow even if it should be a login flow. However since it was identified to be a registered user, the end result is proving the same. So to catch all ends, we have to cover the callbacks.

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthOptions.tsx:198]: Will verify soon which ones we can let go and the ones we should keep.

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (capJavert) [inline packages/shared/src/components/auth/AuthOptions.tsx:72]: `{ isLoginFlow }` to allow futher customization?

File: packages/webapp/pages/onboarding.tsx
Comment (capJavert) [inline packages/webapp/pages/onboarding.tsx:219]: If `isLogin` is false do we then still set `isAuthenticating` to `true`?

**Review by capJavert — COMMENTED** [review]
Few comments for sanity checking, overall looks good. 👍

File: packages/webapp/pages/onboarding.tsx
Comment (sshanzel) [reply packages/webapp/pages/onboarding.tsx]: It can be registration which is still under authenticating, so both are always true for that property.

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthOptions.tsx:73]: Good point. Let's do that.

File: packages/shared/src/components/auth/AuthDefault.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthDefault.tsx:172]: The loading is required indeed. The props should not anymore. Good catch.

File: packages/shared/src/components/modals/common/ModalHeader.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/common/ModalHeader.tsx:45]: All other places are broken actually, this fixes them 😅 

The close button should be relative to the header when you scroll, without this it overlaps the elements.

Will also keep an eye out to see any weird behaviors after this.

Comment (capJavert) [reply packages/shared/src/components/modals/common/ModalHeader.tsx:45]: Oh 😆 nothing to add then 👏

File: packages/shared/src/components/auth/AuthContainer.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthContainer.tsx:5]: Yep. The goal was to have a column, hence, flex-col should be alright.

File: packages/shared/src/components/auth/AuthSignBack.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthSignBack.tsx:41]: Wait this shouldn't be conditionally wrapped right, rather conditionally rendered?
We should not show this part if you are on email login flow?

File: packages/shared/src/components/buttons/Button.tsx
Comment (rebelchris) [inline packages/shared/src/components/buttons/Button.tsx:137]: I still think this is a little bit messy to be honest.
It feels like it abuses the conditional wrapper, what part is this actually needed for?
Maybe we can find a cleaner implementation or revert to rendering it in the component rather.

File: packages/shared/src/hooks/auth/useSignBack.ts
Comment (rebelchris) [inline packages/shared/src/hooks/auth/useSignBack.ts:13]: Why do we use the join here? 👀

Comment (rebelchris) [reply packages/shared/src/hooks/auth/useSignBack.ts:13]: It's a react query key, used for something else, I would suggest either making a new function or finding another solution.
Try to keep things seperate

File: packages/shared/src/components/ConditionalWrapper.tsx
Comment (rebelchris) [inline packages/shared/src/components/ConditionalWrapper.tsx:3]: I think ideally this wouldn't be needed to be exported.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Wow amazing refactor, almost there I think!

File: packages/shared/src/components/auth/AuthSignBack.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthSignBack.tsx:41]: We need the profile picture for both ends. It will need a wrapper conditionally to display the social provider icon.

File: packages/shared/src/components/buttons/Button.tsx
Comment (sshanzel) [reply packages/shared/src/components/buttons/Button.tsx:137]: This was some unreverted changes. This was part of using the button before the custom css. Shall revert this.

File: packages/shared/src/components/auth/AuthSignBack.tsx
Comment (rebelchris) [reply packages/shared/src/components/auth/AuthSignBack.tsx:41]: Oh I see, ok that makes sense, but does it not render completely different?
Like one is in a button, and one is just the big version of profile image?

Comment (sshanzel) [reply packages/shared/src/components/auth/AuthSignBack.tsx:41]: No, just really needs the wrapper with a relative position for the absolute placement of the icon.

File: packages/shared/src/hooks/auth/useSignBack.ts
Comment (sshanzel) [reply packages/shared/src/hooks/auth/useSignBack.ts:13]: This is also a key to be used for storage. What it accepts is a string and to keep the keys uniform, reused the function.

Comment (rebelchris) [reply packages/shared/src/hooks/auth/useSignBack.ts:13]: Maybe we can duplicate the function and name it explicitly and do the join in the function?
Will be more uniform moving forward for storage keys.

WDYT?

File: packages/shared/src/components/auth/AuthSignBack.tsx
Comment (rebelchris) [reply packages/shared/src/components/auth/AuthSignBack.tsx:41]: Ah ok, closing this as resolved then.

File: packages/shared/src/hooks/auth/useSignBack.ts
Comment (sshanzel) [reply packages/shared/src/hooks/auth/useSignBack.ts]: Good point! Let's do that.

**Review by rebelchris — APPROVED** [review]
Alright looks good.
As it's a critical flow as well, please have one more review and perform critical flow tests once merged to main with updated test line.

**Pull request merged**
